src/App.tsx

```diff
@@ -5,7 +5,13 @@ import { FindMovie } from './components/FindMovie';
 import { Movie } from './types/Movie';
 
 export const App = () => {
-  const [movies] = useState<Movie[]>([]);
+  const [movies, setMovies] = useState<Movie[]>([]);
+
+  const addMovie = (newMovie: Movie) => {
+    if (!movies.some(movie => movie.imdbId === newMovie.imdbId)) {
```

Comment:
Would be better to move `!movies.some(movie => movie.imdbId === newMovie.imdbId)` to a variable